src/rust/ensogl/src/display/shape/text/glyph/system.rs

```diff
@@ -74,88 +82,165 @@ impl Glyph {
 
 /// A structure keeping line of glyphs with proper alignment.
 ///
-/// Not all the glyphs in `glyphs` vector may be actually in use; this structure is meant to keep
+/// Not all the glyphs in `glyphs` vector may be actually in use. This structure is meant to keep
 /// changing text, and for best performance it re-uses the created Glyphs (what means the specific
-/// buffer space). Therefore there is a cap for line length. See also `GlyphSystem::new_empty_line`.
-#[derive(Debug)]
+/// buffer space). Therefore you can set a cap for line length by using the `set_fixed_capacity`
+/// method.
+#[derive(Clone,CloneRef,Debug)]
 pub struct Line {
-    glyphs         : Vec<Glyph>,
-    baseline_start : Vector2<f32>,
-    base_color     : Vector4<f32>,
-    height         : f32,
-    font           : FontHandle,
+    display_object : display::object::Instance,
+    glyph_system   : GlyphSystem,
+    content        : Rc<RefCell<String>>,
+    glyphs         : Rc<RefCell<Vec<Glyph>>>,
+    font_color     : Rc<Cell<color::Rgba>>,
+    font_size      : Rc<Cell<f32>>,
+    fixed_capacity : Rc<Cell<Option<usize>>>,
```

Comment:
As discussed via phone, we have assumption that all shape systems are CloneRef. It may appear that it is more optimal to group several fields in a single Rc here, but lets leave that for optimizations time one day.